### DISCUSSION TITLE
How to treat time?

### DISCUSSION THREAD

**Discussion by EvanKomp** [e3nn/e3nn#400] [Q&A]
Hey all! Thanks for any advice in advance. My data are trajectories in 2D space, eg a collection of (time, x, y) pairs. I am trying to determine if this use case is a good application of e3nn.

Specifically:
1. I know that I can just stick 0's on my data as "z" everything is kosher for O(3).
2. I cannot decide if time should be a node feature (scaler) or the 3rd dimension in the distance vector.

It is clear that I want the model to be eq. to rotation, translation, inversion. On the other hand, time should also have eq. to translation... its the difference in time between points that matters. This makes me think time should be a component of the vector, but also, a rotation in (time, x, y) isn't very sensible.

**Comment by blondegeek** [comment]
Like you said, time should not be treated as a spatial coordinate. It would
be a scalar feature. Also if you only have x,y you should use a network
that is E(2) equivariant rather than E(3). You may also want to consider
treating "relative time" between frames rather than absolute if you have
time translation symmetry.

Best,
Tess
